plugins/linguist-backend/src/plugin.ts

```diff
@@ -19,69 +19,79 @@ import {
 import { HumanDuration } from '@backstage/types';
 
-/**
- * Options for Linguist backend plugin
- *
- * @public
- */
-export interface LinguistPluginOptions {
-  schedule?: TaskScheduleDefinition;
-  age?: HumanDuration;
-  batchSize?: number;
-  useSourceLocation?: boolean;
-  linguistJsOptions?: Record<string, unknown>;
-  kind?: string[];
-}
+import { createRouter } from './service/router';
 
 /**
  * Linguist backend plugin
  *
  * @public
  */
-export const linguistPlugin = createBackendPlugin(
-  (options: LinguistPluginOptions) => ({
-    pluginId: 'linguist',
-    register(env) {
-      env.registerInit({
-        deps: {
-          logger: coreServices.logger,
-          reader: coreServices.urlReader,
-          database: coreServices.database,
-          discovery: coreServices.discovery,
-          scheduler: coreServices.scheduler,
-          tokenManager: coreServices.tokenManager,
-          httpRouter: coreServices.httpRouter,
-        },
-        async init({
-          logger,
-          reader,
-          database,
-          discovery,
-          scheduler,
-          tokenManager,
-          httpRouter,
-        }) {
-          httpRouter.use(
-            await createRouter(
-              {
-                ...options,
-              },
-              {
-                logger: loggerToWinstonLogger(logger),
-                reader,
-                database,
-                discovery,
-                scheduler,
-                tokenManager,
-              },
-            ),
+export const linguistPlugin = createBackendPlugin({
+  pluginId: 'linguist',
+  register(env) {
+    env.registerInit({
+      deps: {
+        logger: coreServices.logger,
+        config: coreServices.rootConfig,
+        reader: coreServices.urlReader,
+        database: coreServices.database,
+        discovery: coreServices.discovery,
+        scheduler: coreServices.scheduler,
+        tokenManager: coreServices.tokenManager,
+        httpRouter: coreServices.httpRouter,
+      },
+      async init({
+        logger,
+        config,
+        reader,
+        database,
+        discovery,
+        scheduler,
+        tokenManager,
+        httpRouter,
+      }) {
+        let schedule;
+        if (config.has('linguist.schedule')) {
+          schedule = readTaskScheduleDefinitionFromConfig(
+            config.getConfig('linguist.schedule'),
           );
-        },
-      });
-    },
-  }),
-);
+        }
+        const batchSize = config.getOptionalNumber('linguist.batchSize');
+        const useSourceLocation = config.getBoolean(
+          'linguist.useSourceLocation',
+        );
+        const age = config.getOptionalConfig('linguist.age') as
+          | HumanDuration
+          | undefined;
+        const kind = config.getOptionalStringArray('linguist.kind');
+        const linguistJsOptions = config.getOptionalConfig(
+          'linguist.linguistJsOptions',
+        );
+
+        httpRouter.use(
+          await createRouter(
```

Comment:
I'm thinking it'd be better to move the config reading into `createRouter` instead, that way the config will work with the old system too